test/unit/gotcha_unit_tests.c

```diff
@@ -221,11 +237,85 @@ Suite* gotcha_libc_suite(){
   tcase_add_test(libc_case, gotcha_heap_test);
   tcase_add_test(libc_case, gotcha_memcpy_test);
   tcase_add_test(libc_case, gotcha_strcmp_test);
+  tcase_add_test(libc_case, gotcha_atoi_test);
+  suite_add_tcase(s, libc_case);
+  return s;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////GOTCHA Auxv Tests///////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+START_TEST(vdso_map_test){
+  struct link_map* hopefully_vdso = get_vdso_from_maps();
+  struct link_map* hopefully_also_vdso = get_vdso_from_auxv();
+  // intentionally not testing aliases explicitly as many systems don't implement this
+  ck_assert_msg(hopefully_vdso, "VDSO not found in maps");
+  ck_assert_msg(hopefully_also_vdso, "VDSO not found in auxv");
```

Comment:
This will false-negative fail on some systems, as not every get_vdso_* mechanism works everywhere.  How about: 

struct link_map *maps_vdso = get_vdso_from_maps();
struct link_map *auxv_vdso = get_vdso_from_auxv();
struct link_map *alias_vdso = get_vdso_from_aliases();
ck_assert_msg(maps_vdso || auxv_vdso || alias_vdso, "VDSO not found in any solution");
ck_assert_msg(!maps_vdso || !auxv_vdso || maps_vdso == auxv_vdso, "VDSO Mismatch of maps and auxv");
ck_assert_msg(!maps_vdso || !alias_vdso || maps_vdso == alias_vdso, "VDSO Mismatch of maps and alias");
ck_assert_msg(!auxv_vdso || !alias_vdso || auxv_vdso == alias_vdso, "VDSO Mismatch of auxv and alias");